x-pack/legacy/plugins/ml/public/overview/components/anomaly_detection_panel/utils.ts

```diff
@@ -0,0 +1,174 @@
+            earliest_timestamp: job.earliestTimestampMs,
+            latest_timestamp: job.latestTimestampMs,
+            max_anomaly_score: null,
+          };
+        } else {
+          groups[g].jobIds.push(job.id);
+          groups[g].docs_processed += job.processed_record_count;
+          // if incoming job latest timestamp is greater than the last saved one, replace it
+          if (groups[g].latest_timestamp === undefined) {
+            groups[g].latest_timestamp = job.latestTimestampMs;
+          } else if (job.latestTimestampMs > groups[g].latest_timestamp) {
+            groups[g].latest_timestamp = job.latestTimestampMs;
+          }
+          // if existing group's earliest timestamp is not defined, set it to current job's
+          // otherwise compare to current job's and replace if current job's earliest is less
+          if (groups[g].earliest_timestamp === undefined) {
+            groups[g].earliest_timestamp = job.earliestTimestampMs;
+          } else if (job.earliestTimestampMs < groups[g].earliest_timestamp) {
+            groups[g].earliest_timestamp = job.earliestTimestampMs;
+          }
+        }
+      });
+    } else {
+      groups.ungrouped.jobIds.push(job.id);
+      groups.ungrouped.docs_processed += job.processed_record_count;
+      // if incoming job latest timestamp is greater than the last saved one, replace it
+      if (job.latestTimestampMs > groups.ungrouped.latest_timestamp) {
+        groups.ungrouped.latest_timestamp = job.latestTimestampMs;
+      }
+    }
+  });
+
+  return groups;
+}
+
+export function getStatsBarData(jobsList: any) {
+  const jobStats = {
+    activeNodes: {
+      label: i18n.translate('xpack.ml.overviewJobsList.statsBar.activeMLNodesLabel', {
+        defaultMessage: 'Active ML Nodes',
+      }),
+      value: 0,
+      show: true,
+    },
+    total: {
+      label: i18n.translate('xpack.ml.overviewJobsList.statsBar.totalJobsLabel', {
+        defaultMessage: 'Total jobs',
+      }),
+      value: 0,
+      show: true,
+    },
+    open: {
+      label: i18n.translate('xpack.ml.overviewJobsList.statsBar.openJobsLabel', {
+        defaultMessage: 'Open jobs',
+      }),
+      value: 0,
+      show: true,
+    },
+    closed: {
+      label: i18n.translate('xpack.ml.overviewJobsList.statsBar.closedJobsLabel', {
+        defaultMessage: 'Closed jobs',
+      }),
+      value: 0,
+      show: true,
+    },
+    failed: {
+      label: i18n.translate('xpack.ml.overviewJobsList.statsBar.failedJobsLabel', {
+        defaultMessage: 'Failed jobs',
+      }),
+      value: 0,
+      show: false,
+    },
+    activeDatafeeds: {
+      label: i18n.translate('xpack.ml.jobsList.statsBar.activeDatafeedsLabel', {
+        defaultMessage: 'Active datafeeds',
+      }),
+      value: 0,
+      show: true,
+    },
+  };
+
+  if (jobsList === undefined) {
+    return jobStats;
+  }
+
+  // object to keep track of nodes being used by jobs
+  const mlNodes: any = {};
```

Comment:
Looking at how this is used later on, could this just be an array of `job.nodeName`s like `const mlNodes: string[] = [];` or even just a counter like `let activeNodes = 0;`?